XLS-47d-PriceOracles/README.md

```diff
@@ -0,0 +1,453 @@
+
+|FieldName | Required? | JSON Type | Internal Type|
+|:---------|:-----------|:---------------|:---------------|
+| `LedgerEntryType`| :heavy_check_mark: | `string` | `UINT16` |
+| `Owner` | :heavy_check_mark: | `string` | `ACCOUNTID` |
+| `Provider` | :heavy_check_mark: | `string` | `BLOB` |
+| `PriceDataSeries` | :heavy_check_mark: | `array` | `ARRAY` |
+| `LastUpdateTime` | :heavy_check_mark: | `number` | `UINT32` |
+| `URI` | | `string` | `BLOB` |
+| `SymbolClass` | :heavy_check_mark: | `string` | `BLOB` |
+| `PreviousTxnID` | :heavy_check_mark: | `string` | `UINT256` |
+| `PreviousTxnLgrSeq`| :heavy_check_mark: | `string` | `UINT32` |
+
+- `LedgerEntryType` identifies the type of ledger object. The proposal recommends the value 0x0080 as the reserved entry type.
+- `Owner` is the account that has the update and delete privileges. It is recommended that this account has an associated [`signer list`](https://xrpl.org/set-up-multi-signing.html).
+- `Provider` identifies an Oracle Provider. It can be URI or any data, for instance `chainlink`. It is a string of up to 256 ASCII hex encoded characters (0x20-0x7E).
+- `PriceDataSeries` is an array of up to ten `PriceData` objects, where `PriceData` represents the price information for a token pair. `PriceOracle` with more than five `PriceData` objects requires two owner reserves. `PriceData` includes the following fields:
+
+  |FieldName | Required? | JSON Type | Internal Type|
+  |:---------|:-----------|:---------------|:---------------|
+  | `Symbol` | :heavy_check_mark: | `string` | `CURRENCY` |
+  | `PriceUnit` | :heavy_check_mark: | `string` | `CURRENCY` |
+  | `SymbolPrice` | | `number` | `UINT64` |
+  | `Scale` | | `number` | `UINT8` |
+
+  - `Symbol` is the symbol to be priced. Any arbitrary value should be allowed and interpreted exactly like other asset code fields in the ledger. A new enum value STI_CURRENCY and class STCurrency are introduced to support the `CURRENCY` field.
+  - `PriceUnit` is the denomination in which the prices are expressed. Any arbitrary value should be allowed and interpreted exactly like other asset code fields in the ledger.
+  - `SymbolPrice` is the scaled asset price, which is the price value after applying the scaling factor. This is an optional field. It is not included if the last update transaction didn't include the `Symbol`/`PriceUnit` pair.
+  - `Scale` is the price's scaling factor. It represents the price's precision level. For instance, if `Scale` is `6` and the original price is `0.155` then the scaled price is `155000`. Formally, $scaledPrice = originalPrice*{10}^{scale}$. Valid `Scale` range is {0-10}. This is an optional field. It is not included if the last update transaction didn't include the `Symbol`/`PriceUnit` pair.
+
+- `URI` is an optional URI field to reference the price data off-chain. It is a string of up to 256 ASCII hex encoded characters (0x20-0x7E).
```

Comment:
It's probably enough to just say this is a URI, correct? We can let that spec (e.g., [RFC-3986](https://datatracker.ietf.org/doc/html/rfc3986)) define the types of allowed characters.

If you agree, we could change this to just:

```suggestion
- `URI` is an optional [URI](https://datatracker.ietf.org/doc/html/rfc3986) field to reference price data off-chain. It is limited to 256 bytes.
```